parser/lexer.go

```diff
@@ -520,7 +528,7 @@ func startWithSlash(s *Scanner) (tok int, pos Pos, lit string) {
 
 	case '+': // '/*+' optimizer hints
 		// See https://dev.mysql.com/doc/refman/5.7/en/optimizer-hints.html
-		if _, ok := hintedTokens[s.lastKeyword]; ok {
+		if _, ok := hintedTokens[s.lastKeyword]; ok || s.keepHint {
```

Comment:
Why is this change needed? The check is to mimic MySQL's behavior that a hint must follow certain keywords (SELECT, INSERT, etc.); I don't think normalization will move the hint elsewhere.

Reply:
When normalizing SQL, the value of lastKeyword will always be 0, resulting in the omission of all hints.

Reply:
I am unable to find an appropriate method of handling it.